sbncode/EventGenerator/MeVPrtl/Tools/BNBKaonGen_tool.cc

```diff
@@ -0,0 +1,538 @@
+/**
+ *
+ */
+
+// Framework Includes
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art/Persistency/Common/PtrMaker.h"
+#include "art/Utilities/ToolMacros.h"
+#include "cetlib/cpu_timer.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+#include "CLHEP/Random/RandFlat.h"
+#include "ifdh_art/IFDHService/IFDH_service.h"
+
+#include "nusimdata/SimulationBase/MCFlux.h"
+
+// local includes
+#include "IMesonGen.h"
+#include "boone.h"
+#include "PDGCodes.h"
+
+// LArSoft includes
+#include "dk2nu/tree/dk2nu.h"
+#include "dk2nu/tree/dkmeta.h"
+#include "dk2nu/tree/NuChoice.h"
+
+// ROOT
+#include "TVector3.h"
+#include "TTree.h"
+#include "TFile.h"
+#include "TTreeReader.h"
+#include "TTreeReaderValue.h"
+
+// std includes
+#include <string>
+#include <iostream>
+#include <memory>
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+// implementation follows
+
+namespace evgen {
+namespace ldm {
+/**
+ *  @brief  BNBKaonGen class definiton
+ */
+class BNBKaonGen : public IMesonGen
```

Comment:
At some point it might be worthwhile to refactor BNB/NuMi gens so that they share overlapping code.